src/expression.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any
+from .token_type import TokenType, Token
+
+
+class Expr:
+    def __call__(self) -> Any:
+        raise RuntimeError("Empty Expresson")
+
+
+class Binary(Expr):
+    left: Expr
+    operator: Token
+    right: Expr
+
+    def __init__(self, left: Expr, operator: Token, right: Expr):
+        self.left = left
+        self.operator = operator
+        self.right = right
+
+    def __str__(self) -> str:
+        return f"( {self.operator} {str(self.left)}, {str(self.right)} )"
+
+    def __call__(self) -> Any:
```

Comment:
@SirPaws @spralja @ardaOnal @GangstersPairOfDice any thoughts on whether we should use these built in dunder methods or make it more explicit?

Reply:
@SirPaws Sure :) Maybe this code will make what I mean more clear

```python
class Shit:
  evaluate(self): # Explicit Method
     return 10
  __call__(self): # Dunder Method
     return 10
     
 myShit = Shit()
 myShit() # 10
 myShit.evaluate() # 10
```

The dunder method `__call__` gets mapped to `()` (like if you call the object like a function) so we kind of hide the fact we're evaluating the value through some cool syntax. We do the same thing for making it a string with`__str__` instead of `.stringify()` for exampel

Reply:
Yeah I think they're great